delphi/app/pki_model.py

```diff
@@ -0,0 +1,106 @@
+from delphi.app.graph import run_optimization
+
+
+class PKI:
```

Comment:
This should just be a flow network object

Reply:
I think it is a better idea to not make it into a flow network object. It simplifies updating room counts and enabling/disabling. That is why it was implemented this way. Also, I do not want to track down issues in the flow network class by trying to account for enabled/disabled nodes when optimizing.

Reply:
I'm not sure if it would complicated the flow network that much, since most of what you're doing here could be ported over. The issue I see arising is that you'll be bouncing information between the two classes which makes it difficult to follow especially when we're getting updates about different rooms that needs to be communicated to both. It's using the flow network to do it's calculations, so you could have the flow network keep track of which nodes are enabled/disabled and then only operate on the set of enabled nodes. That way you have all of the information for computing the flow stored on the analyzer for it to grab whenever it gets any new or updated data

Reply:
I'm not a fan of having to do a bunch of unnecessary checking logic to see if nodes are enabled in the flow model. Might as well not even add disabled nodes/edges in the first place since we have to rerun the optimization anyways. And Frankly, we are cutting it close on time anyways, so If there isn't anything inherently wrong with the way it is being done, I propose we leave it.

Reply:
This is making it confusing to follow the code. From my understanding, it seems like you update the values on the PKI object, and then the flow network object gets those values, and then computes, and then util function is used to aggregate those results, which is a very confusing program flow.

Reply:
At this point, I think we both are proposing solutions that work. As it stands, my solution is implemented. We do not have a ton of time left on this project, and your proposed solution will take me some time to implement and test, time that will be hard for me to find on top of working on all the paperwork we still need to get done for this project. Unless @mooncollin  and @mooncollin  also have a problem with the way this is implemented, I'm going to say we still with what we have.